lib-rgbpanel/src/rgbpanel.cpp

```diff
@@ -55,27 +58,45 @@ RgbPanel::RgbPanel(uint32_t nColumns, uint32_t nRows, uint32_t nChain, RgbPanelT
 	m_ptColons = new struct TColon[m_nMaxPosition * m_nMaxLine];
 	assert(m_ptColons != nullptr);
 	SetColonsOff();
+
 }
 
 /**
  * Text
  */
 void RgbPanel::PutChar(char nChar, uint8_t nRed, uint8_t nGreen, uint8_t nBlue) {
-	if (__builtin_expect((static_cast<uint32_t>(nChar) >= cp437_font_size()), 0)) {
+
+	uint32_t font_size = 0;
+	uint32_t font_width = 0;
+	uint32_t font_height = 8;
+	uint32_t font_space = 0;
+
+	if (m_nFontID == 0){
+		font_size = cp437_font_size();
+		font_width = FONT_CP437_CHAR_W;
+	} else if (m_nFontID == 1) {
+		font_size = font_5x8_size();
+		font_width = FONT_5x8_CHAR_W;
+//		font_height = FONT_5x8_CHAR_H;
+		font_space = 1;		
+	}
```

Comment:
font_size, font_width and font_space are static values; it should not be calculated all the times in a heavily called function.

Reply:
font width here depends on the font selected for the line, I can pre-calculate size, but not width?

Reply:
Ok. I am just not in favor calculating the same for each character ;-)

Reply:
I copy and modify your cp437_font_size() for my 5x8, mine is virtually the same,  does this division get optimized away into a single constant, because they are both compile time constants?

```
uint32_t cp437_font_size(void) {
	return sizeof(cp437_font)/sizeof(cp437_font[0]);
}
```